dashboard/src/components/settings/roles/RoleSettings/RoleSettings.tsx

```diff
@@ -0,0 +1,304 @@
+    const originalIndex = availableRoles.findIndex(
+      (role) => role.name === originalRole.name,
+    );
+    const updatedRoles = availableRoles.map((role, index) =>
+      index === originalIndex ? { name, isSystemRole: false } : role,
+    );
+
+    setValue('authUserDefaultAllowedRoles', updatedRoles, {
+      shouldDirty: true,
+    });
+  }
+
+  function handleRemoveRole({ name }: Role) {
+    const filteredRoles = availableRoles.filter((role) => role.name !== name);
+
+    setValue('authUserDefaultAllowedRoles', filteredRoles, {
+      shouldDirty: true,
+    });
+  }
+
+  function handleOpenCreator() {
+    openDialog('MANAGE_ROLE', {
+      title: (
+        <span className="grid grid-flow-row">
+          <span>Add Role</span>
+
+          <Text variant="subtitle1" component="span">
+            Enter the name for the role below.
+          </Text>
+        </span>
+      ),
+      payload: {
+        availableRoles,
+        submitButtonText: 'Create',
+        onSubmit: handleAddRole,
+      },
+      props: { PaperProps: { className: 'max-w-sm' } },
+    });
+  }
+
+  function handleOpenEditor(originalRole: Role) {
+    openDialog('MANAGE_ROLE', {
+      title: (
+        <span className="grid grid-flow-row">
+          <span>Edit Role</span>
+
+          <Text variant="subtitle1" component="span">
+            Enter the name for the role below.
+          </Text>
+        </span>
+      ),
+      payload: {
+        originalRole,
+        availableRoles,
+        onSubmit: (values: RoleFormValues) =>
+          handleEditRole(values, originalRole),
+      },
+      props: { PaperProps: { className: 'max-w-sm' } },
+    });
+  }
+
+  function handleConfirmRemove(originalRole: Role) {
+    openAlertDialog({
+      title: 'Remove Role',
+      payload: (
+        <Text>
+          Are you sure you want to remove the &quot;
+          <strong>{originalRole.name}</strong>&quot; role?
+        </Text>
+      ),
+      props: {
+        onPrimaryAction: () => handleRemoveRole(originalRole),
+        primaryButtonColor: 'error',
+        primaryButtonText: 'Remove',
+      },
+    });
+  }
+
+  async function handleSubmit(values: RoleSettingsFormValues) {
+    const updateAppPromise = updateApp({
+      variables: {
+        id: currentApplication?.id,
+        app: {
+          authUserDefaultAllowedRoles: values.authUserDefaultAllowedRoles
+            .map(({ name }) => name)
+            .join(','),
+        },
+      },
+    });
+
+    await toast.promise(
+      updateAppPromise,
+      {
+        loading: 'Updating roles...',
+        success: 'Roles have been updated successfully.',
+        error: 'An error occurred while updating roles.',
+      },
+      toastStyleProps,
+    );
+  }
+
+  return (
+    <FormProvider {...form}>
+      <Form onSubmit={handleSubmit}>
+        <SettingsContainer
+          title="Roles"
+          description="Add and change permissions for different roles."
```

Comment:
this section is only for adding/updating roles. There are no permissions involved.